sklearn/ensemble/_hist_gradient_boosting/tests/test_splitting.py

```diff
@@ -478,3 +487,199 @@ def test_splitting_missing_values(X_binned, all_gradients,
 
         assert set(samples_right) == set(missing_samples_indices)
         assert set(samples_left) == set(non_missing_samples_indices)
+
+
+@pytest.mark.parametrize(
+    'X_binned, has_missing_values, n_bins_non_missing, ', [
+        # one category
+        ([0] * 20, False, 1),
+
+        # all categories appear less than CAT_SMOOTH
+        ([0] * 9 + [1] * 8, False, 2),
+
+        # only one category appear more than CAT_SMOOTH
+        ([0] * 12 + [1] * 8, False, 2),
+
+        # missing values + category appear less than CAT_SMOOTH
+        # 9 is missing
+        ([0] * 9 + [1] * 8 + [9] * 4, True, 2),
+
+        # 9 is missing
+        ([9] * 11, True, 0),
+    ])
+def test_splitting_categorical_no_splits(X_binned, has_missing_values,
+                                         n_bins_non_missing):
+    # Checks categorical splits are correct when there are no spliits
```

Comment:
Check that the categorical split gain is -1 when there is no possible split

Reply:
With my next push, this is checked at the end of the test.